system/default_ad_api_helpers/ad_api_adaptors/src/routing_adaptor.cpp

```diff
@@ -29,29 +29,57 @@ RoutingAdaptor::RoutingAdaptor() : Node("routing_adaptor")
   const auto adaptor = component_interface_utils::NodeAdaptor(this);
   adaptor.init_cli(cli_route_);
   adaptor.init_cli(cli_clear_);
-  route_points_ = std::make_shared<SetRoutePoints::Service::Request>();
+  adaptor.init_sub(
+    sub_state_, [this](const RouteState::Message::ConstSharedPtr msg) { state_ = msg->state; });
+
+  const auto rate = rclcpp::Rate(5.0);
+  timer_ = rclcpp::create_timer(
+    this, get_clock(), rate.period(), std::bind(&RoutingAdaptor::on_timer, this));
+
+  state_ = RouteState::Message::UNKNOWN;
+  route_ = std::make_shared<SetRoutePoints::Service::Request>();
 }
 
-void RoutingAdaptor::on_goal(const PoseStamped::ConstSharedPtr pose)
+void RoutingAdaptor::on_timer()
 {
-  route_points_->header = pose->header;
-  route_points_->goal = pose->pose;
-  route_points_->waypoints.clear();
+  // Wait a moment to merge waypoints. This value is used for flag and delay counter.
+  constexpr int delay_count = 3;
+  if (0 < request_control_ && request_control_ < delay_count) {
+    ++request_control_;
+  }
+  if (request_control_ != delay_count) {
+    return;
+  }
```

Comment:
I don't understand this algorithm well.
Why 3 is chosen?

What does `merge waypoints` mean.

`request_control_` seems to have additional meaning, not only what the variable name expresses.
Could you write a comment of what `request_control_` means.

Reply:
Done https://github.com/autowarefoundation/autoware.universe/pull/2215/commits/343a3572982dbbabbde4035b279d937b8b816e10.